public/scripts/openai.js

```diff
@@ -1705,6 +1713,15 @@ async function sendOpenAIRequest(type, messages, signal) {
             throw new Error(data);
         }
 
+        if (type !== 'quiet') {
+            const logprobs = getLogprobsData(data);
+            // `saveLogprobs` needs the message ID to save the logprobs to the
+            // correct message, but for non-streaming requests the message will
+            // not have been created until after this function returns, hence
+            // the delay.
+            delay(1).then(() => saveLogprobs(logprobs, null));
+        }
```

Comment:
I don't feel great about this approach but saveLogprobs needs to know which message to associate the logprobs with, and the message doesn't exist yet.  Ideally this would happen in `maybeSaveLogprobs` in script.js at the very end of `Generate`, but that would require modifying the function signature for all of the Chat Completion non-streaming request handlers to return `{text:string, logprobs: TokenLogprobs[]}` instead of `string`. There's a lot of those handlers and no types so that seems like too risky of a change considering I can't personally test most of the chat completion handlers due to lack of access.